### PR TITLE
Remove ReactiveReSwift

### PR DESCRIPTION
**Pull request by onmyway133**
- Remove ReactiveReSwift, use just ReSwift + RxSwift
- Add `Subscriber` as a reactive way to subscribe to store, it works like. It automatically ubsubscribes from `Store` when `deinit`

```swift
let store = Store(reducer: Reducer(), state: State(value: 0))
let subscriber = Subscriber(store: store)

subscriber.state.asObservable().subscribe ...
```

### CODE REVIEW

File: Sources/Shared/Navigation/Navigator.swift [hunk 1]
```diff
@@ -42,8 +42,9 @@ public final class Navigator {
     self.currentController = currentController
     Compass.routes = Array(navigationRouter().routes.keys)
 
+    // FIXME
     if let intentRouter = intentRouter?() {
-      Array(intentRouter.routes.keys)
+      let _ = Array(intentRouter.routes.keys)
```
Comment (vadymmarkov) [inline Sources/Shared/Navigation/Navigator.swift]: Oh, I forgot about this one, it should be `Compass.routes += Array(intentRouter.routes.keys)`

File: Sources/Shared/Promise.swift [hunk 2]
```diff
@@ -27,7 +27,7 @@ extension Promise: ObservableConvertibleType {
 
 // MARK: - JSON dictionary
 
-extension Collection where Self: DictionaryLiteralConvertible, Self.Key == String, Self.Value: Any, Iterator.Element == (key: Self.Key, value: Self.Value) {
+extension Collection where Self: ExpressibleByDictionaryLiteral, Self.Key == String, Self.Value: Any, Iterator.Element == (key: Self.Key, value: Self.Value) {
```
Comment (vadymmarkov) [inline Sources/Shared/Promise.swift:30]: @onmyway133 Does it change something?

File: Sources/Shared/Navigation/Navigator.swift (hunk 1 above)
Comment (onmyway133) [reply Sources/Shared/Navigation/Navigator.swift]: I can add this

File: Sources/Shared/Promise.swift (hunk 2 above)
Comment (onmyway133) [reply Sources/Shared/Promise.swift:30]: They are just type alias https://developer.apple.com/reference/swift/dictionaryliteralconvertible, but somehow the compiler shows some warnings

**Comment by vadymmarkov** [comment]
@onmyway133 Looks great! 🚀

**Pull request merged**